fplll/pruner/pruner.h

```diff
@@ -290,37 +386,135 @@ template <class FT> class Pruner
       symmetry_factor = 1;
     }
     min_pruning_coefficients.resize(d);
+    btmp.resize(d);
+    bftmp.resize(n);
     fill(min_pruning_coefficients.begin(), min_pruning_coefficients.end(), 0.);
-
     set_tabulated_consts();
+
+    // optimize overall cost, use
+    if (flags & PRUNER_SINGLE)
+    {
+      opt_single = true;
+    }
+
+    // need to fix target if possible
+    if (metric == PRUNER_METRIC_PROBABILITY_OF_SHORTEST)
+    {
+      // check if the target is reasonable; otherwise set it to 0.99.
+      // note the target may affect the cost in target_function().
+      if (this->target > 1.0 || this->target < 0.0)
+      {
+        this->target = 0.99;
+      }
+    }
+    else if (metric == PRUNER_METRIC_EXPECTED_SOLUTIONS)
+    {
+      // check if the target is reasonable; otherwise set it to 0.99.
+      // note the target is allowed to be larger than 1 in this case.
+      // Also the target may affect the cost in target_function().
+      if (this->target < 0.0)
+      {
+        this->target = 0.99;
+      }
+    }
+    else
+    {
+      throw std::invalid_argument("Pruner was set to an unknown metric");
+    }
     load_basis_shapes(gso_rs);
   }
 
   /**
-     Run the optimization process, successively using the algorithm activated
+     @brief Main interface to optimize pruning coefficients
+
+     Main interface to optimize pruning coefficients.
+     It will in turn invoke either of the two functions:
+      (1)   optimize_coefficients_cost_vary_prob() or
+      (2)   optimize_coefficients_cost_fixed_prob()
+     depending on the input "target".
 
-     @brief run the optimization process
+     If the target is negative (e.g. -1), it calls function (1)
```

Comment:
I think this documentation needs updating

Reply:
yep.